samples/PersistenceSample/main.cpp

```diff
@@ -0,0 +1,179 @@
+
+    std::shared_ptr<NovelRT::Ecs::Graphics::DefaultRenderingSystem> renderingSystem =
+        scheduler.GetRegisteredIEcsSystemAs<NovelRT::Ecs::Graphics::DefaultRenderingSystem>();
+
+    NovelRT::Threading::FutureResult<NovelRT::Ecs::Graphics::TextureInfo> textureFuture =
+        renderingSystem->GetOrLoadTexture("novel-chan");
+
+    renderingSystem->ForceVertexTextureFutureResolution();
+
+    auto transformBuffer = scheduler.GetComponentCache().GetComponentBuffer<TransformComponent>();
+    auto entityGraphBuffer = scheduler.GetComponentCache().GetComponentBuffer<EntityGraphComponent>();
+    auto testStructBuffer = scheduler.GetComponentCache().GetComponentBuffer<TestStruct>();
+
+    EntityId parentEntity =
+        renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(), scheduler);
+
+    EntityId childEntity =
+        renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(), scheduler);
+
+    EntityId childOfChildEntity =
+        renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(), scheduler);
+
+    transformBuffer.PushComponentUpdateInstruction(
+        0, childEntity,
+        TransformComponent{NovelRT::Maths::GeoVector3F(200, 200, 0), NovelRT::Maths::GeoVector2F::zero(), 0});
+    transformBuffer.PushComponentUpdateInstruction(
+        0, childOfChildEntity,
+        TransformComponent{NovelRT::Maths::GeoVector3F(200, 200, 0), NovelRT::Maths::GeoVector2F::zero(), 0});
+    entityGraphBuffer.PushComponentUpdateInstruction(0, childEntity, EntityGraphComponent{true, parentEntity, 0});
+    entityGraphBuffer.PushComponentUpdateInstruction(0, childOfChildEntity, EntityGraphComponent{true, childEntity, 0});
+
+    testStructBuffer.PushComponentUpdateInstruction(0, parentEntity, TestStruct{2, 2, false});
+    testStructBuffer.PushComponentUpdateInstruction(0, childEntity, TestStruct{3, 5, false});
+    testStructBuffer.PushComponentUpdateInstruction(0, childOfChildEntity, TestStruct{4, 10, false});
+
+    NovelRT::Timing::Timestamp secondsPassed(0);
+    Chapter chapterToLoad;
+
+    scheduler.RegisterSystem([&](auto delta, auto catalogue) {
+        secondsPassed += delta;
+
+        if (secondsPassed >= NovelRT::Timing::Timestamp::fromSeconds(3))
+        {
+            secondsPassed = NovelRT::Timing::Timestamp(0);
+            chapterToLoad = Chapter::FromEcsInstance(scheduler.GetComponentCache());
+            resourceManagementProvider->GetResourceLoader()->SavePackage("MyChapter.chapter",
+                                                                         chapterToLoad.ToFileData());
+            chapterToLoad.LoadFileData(
+                resourceManagementProvider->GetResourceLoader()->LoadPackage("MyChapter.chapter"));
+            // chapterToLoad.ToEcsInstance(scheduler.GetComponentCache()); IF YOU WANT TO SEE THE FILE SIZE GROW
+            // FOREVER, UNCOMMENT THIS!
```

Comment:
Yes please  : ^ )

Really though, should this be called out in the documentation for that particular function?

Reply:
No - this is just because of how we're using it in the sample.

Reply:
Yikes.